Sources/Basics/VFSOverlay.swift

```diff
@@ -15,37 +15,75 @@ import struct TSCBasic.AbsolutePath
 import protocol TSCBasic.FileSystem
 
 public struct VFSOverlay: Encodable {
-    public struct File: Encodable {
-        enum CodingKeys: String, CodingKey {
+
+    public class Resource: Encodable {
+        private let name: String
+        private let type: String
```

Comment:
nit/optional: Could consider making type an enum.

Reply:
At first I did, but it seemed more complicated to encode into the desired format. Do you see a clear path to that?

Reply:
Experimented a little more an it requires more custom encoding logic so I'll keep this as is.